### DISCUSSION TITLE
Select API equivalent of with_entities?

### DISCUSSION THREAD

**Discussion by bigjools** [sqlalchemy/sqlalchemy#6854] [Usage Questions] [answered]
One of the patterns I make extensive use of in my project with the Query-based API is passing around a basic Query object and then having functions conditionally modify it or use it for subqueries.

The latter is the one that I cannot get working properly in the Select API. I changed a generic Query to a Select and I'm trying to use that as a subquery in another place, however because it has done a select on the model class instead of a single column, there's an error "sqlite3.OperationalError: sub-select returns 11 columns - expected 1".

With the Query API I could just put in `with_entities`, but now it seems that the select columns have to be defined up front. Is there anything I can do?

Thank you.

**Comment by zzzeek** [comment, marked as answer]
the equivalent on select() is [with_only_columns](https://docs.sqlalchemy.org/en/14/core/selectable.html?highlight=with_only_columns#sqlalchemy.sql.expression.Select.with_only_columns), and in the next SQLAlchemy version it will also have some new features too.

**Reply by bigjools** [reply]
Ah thank you, not sure how I missed that.
